android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityAndroidPayFragment.java

```diff
@@ -36,12 +105,89 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
         Bundle bundle = getArguments();
         if (bundle != null) {
-            unityDelegateObjectName = bundle.getString(EXTRA_UNITY_DELEGATE_OBJECT_NAME);
             cart = bundle.getParcelable(EXTRA_PAY_CART);
             countryCode = bundle.getString(EXTRA_COUNTRY_CODE);
+            androidPayEnvironment = bundle.getInt(EXTRA_ANDROID_PAY_ENVIRONMENT,
+                    WalletConstants.ENVIRONMENT_TEST);
+            publicKey = bundle.getString(EXTRA_PUBLIC_KEY);
         }
 
         // Don't recreate this fragment during configuration change.
         setRetainInstance(true);
+
+        currentCheckoutState = CheckoutState.READY;
+        googleApiClient = new GoogleApiClient.Builder(this.getActivity())
+            .addApi(Wallet.API, new Wallet.WalletOptions.Builder()
+                    .setEnvironment(androidPayEnvironment)
+                    .setTheme(WalletConstants.THEME_DARK)
+                    .build())
+            .addConnectionCallbacks(this)
+            .build();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        googleApiClient.connect();
+    }
+
+    @Override
+    public void onStop() {
+        googleApiClient.disconnect();
+        super.onStop();
+    }
+
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+        if (currentCheckoutState == CheckoutState.READY) {
```

Comment:
Added a state enum to check to see if we've already called for a request to get a wallet. In cass of a reconnecting google client in the case of backgrounding the app I hope to use this state to continue where we left off instead of making a new masked wallet request.